### PR TITLE
UGENE-7617 Unable to save dotplot when removing sequence

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

**Review by yalgaer — CHANGES_REQUESTED** [review]
This proposed solution is not the best option.
The best option is not to loose DP and support saving.

In order to save DP when any of the sequence is removed you will need to cache sequence length & name only.

**Review by yalgaer — CHANGES_REQUESTED** [review]
What if the sequence was renamed? We need to listen for the sequence modification events and update cached details.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 1]
```diff
@@ -2898,6 +2900,25 @@ GUI_TEST_CLASS_DEFINITION(test_7616) {
...
+    GTUtilsDialog::waitForDialog(os, new MessageBoxDialogFiller(os, QMessageBox::Yes));
+
+    //2. Click "Remove sequence" (gray cross) on sequence widget toolbar
+    //3. Answer "Yes" in "Save dot-plot" dialog and choose valid path
+    //Expected state: no crash, no errors in the log
+    GTLogTracer lt;
+    auto toolbar = GTWidget::findWidget(os, "views_tool_bar_human_T1 (UCSC April 2002 chr7:115977709-117855134)");
+    GTWidget::click(os, GTWidget::findWidget(os, "remove_sequence", toolbar));
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    CHECK_SET_ERR(!lt.hasErrors(), "Errors in log: " + lt.getJoinedErrorString());
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp:2919]: Add a check that the file was saved and is not empty.

**Comment by yalgaer** [comment]
> What if the sequence was renamed? We need to listen for the sequence modification events and update cached details.

Please ignore this comment. There is no problem with names.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp (hunk 1 above)
Comment (rasputinkirill) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp:2919]: done

**Review by yalgaer — APPROVED** [review]

**Pull request merged**